packages/expo/hooks/lib/upload-source-maps.js

```diff
@@ -0,0 +1,54 @@
+const { upload } = require('bugsnag-sourcemaps')
+const { promisify } = require('util')
+const { tmpdir } = require('os')
+const { sep, join } = require('path')
+const { mkdtemp, writeFile } = require('fs')
+
+module.exports = async (
+  apiKey,
+  iosManifest,
+  iosBundle,
+  iosSourceMap,
+  androidManifest,
+  androidBundle,
+  androidSourceMap,
+  projectRoot
+) => {
+  const dir = await makeTmpDir()
+
+  const androidBundlePath = join(dir, 'android.bundle.js')
+  const androidSourceMapPath = join(dir, 'android.bundle.js.map')
+
+  const iosBundlePath = join(dir, 'ios.bundle.js')
+  const iosSourceMapPath = join(dir, 'ios.bundle.js.map')
+
+  await promisify(writeFile)(iosSourceMapPath, iosSourceMap, 'utf-8')
+  await promisify(writeFile)(iosBundlePath, iosBundle, 'utf-8')
+
+  await promisify(writeFile)(androidSourceMapPath, androidSourceMap, 'utf-8')
+  await promisify(writeFile)(androidBundlePath, androidBundle, 'utf-8')
+
+  // android
+  console.log(`Uploading Android source map`)
+  await upload({
+    apiKey,
+    appVersion: androidManifest.version,
+    minifiedUrl: '*/cached-bundle-experience-*',
```

Comment:
Android stack frames don't contain the CDN url, as the bundle is downloaded and saved to disk at a path which includes `/cached-bundle-experience`.